### ISSUE TITLE
mpv: undefined symbol: archive_read_support_format_rar5

### ISSUE THREAD

**Issue by the-solipsist** [mpv-player/mpv#7503]
### Important Information

Provide following Information:
- mpv version
mpv/bionic,now 2:0.32.0+git~bionic1 amd64

- libarchive version
libarchive13/bionic,now 3.4.0-1.1~bionic amd64

- Linux Distribution and Version
Ubuntu 18.04 (Bionic Beaver)

- Source of the mpv binary
@mc4man's repo:
http://ppa.launchpad.net/mc3man/mpv-tests/ubuntu/pool/main/m/mpv/

This is the source of the libarchive binary as well:
http://ppa.launchpad.net/mc3man/mpv-tests/ubuntu/pool/main/liba/libarchive/

- If known which version of mpv introduced the problem
0.32

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Run `mpv` on the command-line.

### Expected behavior

mpv starts up.

### Actual behavior
Error:
`mpv: symbol lookup error: mpv: undefined symbol: archive_read_support_format_rar5`

I have the version 3.4.0 of libarchive, which as per this [issue](https://github.com/mpv-player/mpv/issues/7334) should have support for rar5.  So I'm not sure why I'm still getting this error.

**Comment by CounterPillow** [comment]
Looks like a broken build to me. Do a `ldd $(which mpv) | grep libarchive` to find what libarchive it's loading. If that libarchive doesn't contain the symbol in its symbol table (check with `objdump -T file/name/here.so | grep rar5`) then it's officially not our problem.

**Comment by mc4man** [comment]
Not sure what your issue is, tested this for quite  some time locally before uploading.
See no issue with ppa supplied packages..
```
ldd /usr/bin/mpv |grep libarchive  libarchive.so.13 => /usr/lib/x86_64-linux-gnu/libarchive.so.13 (0x00007f7e2b7a0000)
objdump -T /usr/lib/x86_64-linux-gnu/libarchive.so.13 | grep rar5
000000000004f8a0 g    DF .text	0000000000000157  Base        archive_read_support_format_rar5 ```

**Comment by the-solipsist** [comment]
> Looks like a broken build to me. 

You're right.

> `ldd $(which mpv) | grep libarchive` 

	libarchive.so.13 => /usr/local/lib/libarchive.so.13 (0x00007fbcf3f86000)

> `objdump -T /usr/local/lib/libarchive.so.13 | grep rar5`

No support for rar5.  @mc4man, given that this version of libarchive is from your PPA repo, is this something you could look into?

**Comment by CounterPillow** [comment]
>/usr/local/lib/

I doubt that version is from his PPA. Unless his PPA installs shit into /usr/local, which I know it doesn't because you can open the .deb as an archive and check inside data.tar.xz and see that the path it uses is /usr/lib so you locally installed some other libarchive from somewhere.

TL;DR your system is broken and you're at fault for this, not mc4man.

**Comment by mc4man** [comment]
That is not my libarchive13, mine is installed to /usr
You have installed a version by some other method to /usr/local

**Comment by the-solipsist** [comment]
Update:
`objdump -T /usr/lib/x86_64-linux-gnu/libarchive.so.13 | grep rar5`
shows that libarchive.so.13 *has* rar5 support.

But checking `ldd mpv` shows that `/usr/local/lib/libarchive.so.13` is being used.

I checked what libarchive13 is installed using
```
sudo apt policy libarchive13
libarchive13:
  Installed: 3.4.0-1.1~bionic
  Candidate: 3.4.0-1.1~bionic
  Version table:
 *** 3.4.0-1.1~bionic 500
        500 http://ppa.launchpad.net/mc3man/mpv-tests/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
     3.2.2-3.1ubuntu0.5 500
        500 http://in.archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages
     3.2.2-3.1 500
        500 http://in.archive.ubuntu.com/ubuntu bionic/main amd64 Packages
```
Not sure how to fix this, and force the right libarchive13 to be used by mpv.

**Comment by the-solipsist** [comment]
**Resolved**
I ksimply renamed `/usr/local/lib/libarchive.so.13`, and now everything is working. Sorry for clogging up the issues tracker.  I hope this benefits someone else who might run into the same problem.  Thanks for your help, @CounterPillow and @mc4man!

**Comment by mc4man** [comment]
Well it's likely you installed libarchive from source, best would be to browse /usr/local as root and delete all files from  that  libarchive , i.e,  /usr/local/lib.. & maybe also /usr/local/include.

**Comment by rubyFeedback** [comment]
I just ran into the same error and found this here via googling.

I had two libarchive variants: one in /usr/lib/ and the other one in /usr/lib/x86 something.

I removed the latter, then recompiled mpv just to make sure (latest git checkout).

I used waf for now, although I noticed meson may be supported too. If anyone has 
time, it would be nice to test compile both, waf variant and meson IF waf is retained
for the time being, to see whether there are any differences in the resulting end
binary. Ideally the build system can detect libarchive not working - perhaps anyone
with meson information can give this a try, but it is not a huge issue per se IMO;
just the prior libarchive missing symbol. At the least people can find the thread
here about this. :)
